web/api/web_api_v1.c

```diff
@@ -615,12 +615,67 @@ inline int web_client_api_request_v1_registry(RRDHOST *host, struct web_client *
     }
 }
 
+inline void web_client_api_request_v1_info_summary_alarm_statuses(const RRDHOST *host, BUFFER *wb) {
+    int alarm_normal = 0, alarm_warn = 0, alarm_crit = 0;
+    RRDCALC *rc;
+    for(rc = host->alarms; rc ; rc = rc->next) {
```

Comment:
For this to be thread safe, you need to `rrdhost_rdlock()`.

Reply:
I've added it.